.github/workflows/package-prepare-release.yml

```diff
@@ -0,0 +1,189 @@
+name: "[Package] Prepare release"
+on:
+  workflow_dispatch:
+    inputs:
+      package:
+        type: choice
+        options:
+        - opentelemetry-propagator-aws-xray
+        - opentelemetry-resource-detector-azure
+        - opentelemetry-sdk-extension-aws
+        description: 'Package to be released'
+        required: true
+
+jobs:
+  prereqs:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.verify.outputs.version }}
+      changelog: ${{ steps.verify.outputs.changelog }}
+      version_file: ${{ steps.verify.outputs.version_file }}
+      release_branch_name: ${{ steps.verify.outputs.release_branch_name }}
+      next_version: ${{ steps.verify.outputs.next_version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: verify
+        name: Verify prerequisites
+        run: |
+          if [[ $GITHUB_REF_NAME != main ]]; then
+            echo this workflow should only be run against main
+            exit 1
+          fi
+
+          path=./$(./scripts/eachdist.py find-package --package ${{ inputs.package }})
+          changelog=$path/CHANGELOG.md
+
+          if ! grep --quiet "^## Unreleased$" $changelog; then
+            echo the change log is missing an \"Unreleased\" section
+            exit 1
+          fi
+
+          version_dev=$(./scripts/eachdist.py version --package ${{ inputs.package }})
+
+          if [[ ! $version_dev =~ ^([0-9]+)\.([0-9]+)[\.|b]{1}([0-9]+).*.dev$ ]]; then
+            echo "unexpected version: $version"
+            exit 1
+          fi
+
+          version=${version_dev%.dev}
+
+          version_file=$(find $path -type f -name "version.py")
+          file_count=$(echo "$version_file" | wc -l)
+
+          if [ "$file_count" -ne 1 ]; then
```

Comment:
Nit: Should we verify that there is a CHANGELOG as well?

Reply:
Added on line ~37